feat: added context redirection from panels to explorer pages

SagarRajput-7 opened the pull request:
### Summary

Context redirection from panels 
-----------------

> **Kind:**
> Specific coordinate (sc): add a button onclick of a specific coordinate

> **Basic Approach:**
> SC:
>        - add the query filters and the specific coordinate (line/bar etc.) as a filter for e.g. service.name = frontend
>        - add the specific timestamp (1 min range)
> 


    
#### Related Issues / PR's

- https://github.com/SigNoz/signoz/issues/3741


#### Screenshots



Coordinate Redirection:



https://github.com/user-attachments/assets/d88962b5-f9b3-4964-a093-5eb2c14d7ee0







#### Affected Areas and Manually Tested Areas

<!-- ✍️ Add details of blast radius and dev testing areas where applicable-->

<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds context redirection from panels to explorer pages with a button on specific coordinates, enhancing navigation with pre-applied filters and time range.
> 
>   - **Behavior**:
>     - Adds context redirection from panels to explorer pages by adding a button on specific coordinates.
>     - Integrates `useNavigateToExplorer` hook for navigation with filters and time range.
>     - Handles graph clicks to show a menu with options to view traces or logs.
>   - **Components**:
>     - Updates `CeleryTaskDetail`, `CeleryTaskBar`, and `CeleryTaskLatencyGraph` to use `useNavigateToExplorer`.
>     - Modifies `WidgetGraphComponent` and `FullView` to handle graph clicks and set current graph reference.
>   - **Utilities**:
>     - Introduces `useGraphClickToShowButton` for handling graph click interactions.
>     - Adds `useNavigateToExplorerPages` for building filters and navigating to explorer pages.
>     - Implements `handleGraphClick` in `utils.ts` for processing graph click events.
>   - **Styles**:
>     - Adds styles for the new button and menu in `GridCardLayout.styles.scss` and `styles.scss`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for cf6822fb20cc1bd25c3ff8b451bda058dbcfc831. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

SagarRajput-7 commented on frontend/src/container/GridCardLayout/GridCard/WidgetGraphComponent.tsx:
```diff
@@ -230,6 +252,47 @@ function WidgetGraphComponent({
+	const createFilterFromData = (
+		data: Record<string, unknown>,
+	): TagFilterItem[] => {
+		const uuidv4 = v4();
```
done

SagarRajput-7 commented on frontend/src/container/GridCardLayout/GridCard/WidgetGraphComponent.tsx:
```diff
@@ -230,6 +236,57 @@ function WidgetGraphComponent({
+				inFocusOrNot: queryData?.inFocusOrNot || false,
+			},
+			navigateRequestType: 'specific',
+		}).then((result) => {
```
done

SagarRajput-7 commented on frontend/src/container/GridCardLayout/useNavigateToExplorerPages.ts:
```diff
@@ -0,0 +1,159 @@
+				(groupFilter) => groupFilter.key?.key === filter?.key?.key,
+			),
+	);
+	console.log('existingFilters', existingFilters, groupByFilters, uniqueFilters);
```
done

SagarRajput-7 commented on frontend/src/components/CeleryTask/useNavigateToExplorer.ts:
```diff
@@ -0,0 +1,104 @@
+					widgetConfig: {
+						query: preparedQuery,
+						panelTypes: PANEL_TYPES.TIME_SERIES,
+						timePreferance: 'GLOBAL_TIME',
```
its seems are many instance of this older also are spelled this only, will check this later

ahmadshaheer commented on frontend/src/components/CeleryTask/useNavigateToExplorer.ts:
```diff
@@ -0,0 +1,104 @@
+
+			let preparedQuery = prepareQuery(filters, dataSource);
+
+			if (shouldResolveQuery) {
+				preparedQuery = await getUpdatedQuery({
+					widgetConfig: {
+						query: preparedQuery,
+						panelTypes: PANEL_TYPES.TIME_SERIES,
+						timePreferance: 'GLOBAL_TIME',
+					},
+					selectedDashboard,
+				});
+			}
```
curious what will happen if the getUpdatedQuery fails.

SagarRajput-7 replied:
handled it, thanks, good catch

SagarRajput-7 commented:
> @SagarRajput-7, as I checked, the graph action menu doesn't display go to logs/traces. can you please check once?
> 
>  2025-03-19.16-45-06.mov

@ahmadshaheer - forgot to update the description, we removed this functionality in one the later commit. Updated the PR description

ahmadshaheer commented on frontend/src/container/GridCardLayout/GridCardLayout.styles.scss:
```diff
@@ -291,6 +291,38 @@
 	}
 }
 
+.view-onclick-show-button {
+	position: absolute;
+	background: var(--bg-vanilla-100);
+	border: 2px solid var(--bg-vanilla-300);
+	border-radius: 4px;
+	cursor: pointer;
+	font-size: 12px;
+	font-weight: 500;
+	box-shadow: none;
+	list-style-type: none;
+	padding: 4px;
+	color: var(--bg-ink-100);
+	display: flex;
+	flex-direction: column;
+	gap: 2px;
+	overflow: auto;
+	max-height: 150px;
+	width: max-content;
+	max-width: 200px;
+
+	.menu-item {
+		cursor: pointer;
+		padding: 4px;
+		padding-right: 12px;
+
+		&:hover {
+			background-color: var(--bg-vanilla-200);
+			border-radius: 2px;
+		}
+	}
+}
```
can we apply the same style as service details graph click? https://www.dropbox.com/scl/fi/f97mfs2tvy8a7z2g4r1b4/Screenshot-from-2025-03-19-16-43-02.png?rlkey=n450bv2xg7o0f3m7nvbpyj89n&dl=0

SagarRajput-7 replied:
sure, will check and will have to pick this separately
